src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -15790,12 +15795,14 @@ static void Report<T>(MyCollection<T> c)
                     expectedOutput: IncludeExpectedOutput("[], [1, 2, 3], "));
                 verifier.VerifyIL("Program.Empty<T>", """
                     {
-                      // Code size       16 (0x10)
+                      // Code size       15 (0xf)
                       .maxstack  1
-                      IL_0000:  call       "T[] System.Array.Empty<T>()"
-                      IL_0005:  newobj     "System.ReadOnlySpan<T>..ctor(T[])"
-                      IL_000a:  call       "MyCollection<T> MyCollectionBuilder.Create<T>(scoped System.ReadOnlySpan<T>)"
-                      IL_000f:  ret
+                      .locals init (System.ReadOnlySpan<T> V_0)
```

Comment:
Do we understand why the new codegen involves an additional local and IL instruction?

Reply:
initobj expects an address on the stack. In some cases a new local is introduced in order to provide that.